CHANGES/2296.bugfix

```diff
@@ -0,0 +1 @@
+Substantial improvements to the memory consumption of syncs, with a modest improvement in time required to sync.
```

Comment:
It might be difficult to summarize that succinctly.  I can maybe add some links to the attached issue, so they can follow the breadcrumb trail more easily if they want to.